pkg/chunk/local/fs_object_client.go

```diff
@@ -64,8 +64,8 @@ func (f *FSObjectClient) GetObject(ctx context.Context, objectKey string) (io.Re
 
 // PutObject into the store
 func (f *FSObjectClient) PutObject(ctx context.Context, objectKey string, object io.ReadSeeker) error {
-	fullPath := path.Join(f.cfg.Directory, objectKey)
-	err := util.EnsureDirectory(path.Dir(fullPath))
+	fullPath := filepath.Join(f.cfg.Directory, objectKey)
+	err := util.EnsureDirectory(filepath.Dir(fullPath))
```

Comment:
Logic in `EnsureDirectory` seems to copy logic in `os.MkdirAll`, I think it can be simplified to 

```
// EnsureDirectory makes sure directory is there, if not creates it if not
func EnsureDirectory(dir string) error {
	return os.MkdirAll(dir, 0777)
}
```

at which point, I am not sure it's really needed.

Reply:
That code was already there added by Tom in the initial commit to add boltdb support. I had just moved that code to util so not sure why it was added in the first place. See https://github.com/cortexproject/cortex/commit/78e83530e03c0fcc729a3e7c640b39e36ce787c7#diff-f0b676dad4198715e462b2faa06cfaa8R220
It is being used at a couple of places so I think it needs a separate PR to address that.